File: zig/src/Event.zig
const Suit = @import("Card.zig").Suit;

pub const Event = enum {
	res,
	tax,
	static_conversion,
};

pub const SuitFlags = struct {
	cute: bool = false,
	dumb: bool = false,
	mal: bool = false,
	played_sentinel: bool = false,
};

pub const NewEvent = struct {
	event: Event = undefined,
	source: []const u8 = "",
	target: []const u8 = "",
	points: f32 = 0,
	suit: Suit = undefined,
};

pub const ResultPlayer = struct {
	player: []const u8 = "",
	cute: f32 = 0,
	dumb: f32 = 0,
	malicous: f32 = 0,
	static_pts: f32 = 0,
	suit_wins: i32 = 0,
	total: f32 = 0,
	played_peek: bool = false,
	played_swap: bool = false,
	peek_flags: SuitFlags = .{},
};

pub const SubmittedHandsResult = struct {
	p1: ResultPlayer = .{},
	p2: ResultPlayer = .{},
	events: []const NewEvent = &.{},
};

File: Event.go
package main

type Event string

const (
	EventRes Event = "res"
	EventTax Event = "tax"
	EventStatic_conversion Event = "static_conversion"
)

type NewEvent struct {
	Event Event `json:"event"`
	Source string `json:"source"`
	Target string `json:"target"`
	Points float64 `json:"points"`
	Suit Suit `json:"suit"`
}

type ResultPlayer struct {
	Player string `json:"player"`
	Cute float64 `json:"cute"`
	Dumb float64 `json:"dumb"`
	Malicous float64 `json:"malicous"`
	Static_pts float64 `json:"static_pts"`
	Suit_wins int `json:"suit_wins"`
	Total float64 `json:"total"`
	Played_peek bool `json:"played_peek"`
	Played_swap bool `json:"played_swap"`
}

type SubmittedHandsResult struct {
	P1 ResultPlayer `json:"p1"`
	P2 ResultPlayer `json:"p2"`
	Events []NewEvent `json:"events"`
}

